site.srht.shrimple.ouch.json

```diff
@@ -0,0 +1,90 @@
+{
+	"id": "site.srht.shrimple.ouch",
+	"runtime": "org.gnome.Platform",
+	"runtime-version": "47",
+	"sdk": "org.gnome.Sdk",
+	"sdk-extensions": [
+		"org.freedesktop.Sdk.Extension.rust-stable"
+	],
+	"command": "ouch",
+	"finish-args": [
+		"--share=network",
+		"--share=ipc",
+		"--socket=fallback-x11",
+		"--device=dri",
+		"--socket=wayland",
+		"--filesystem=home"
+	],
+	"build-options": {
+		"append-path": "/usr/lib/sdk/rust-stable/bin",
+		"build-args": [
+			"--share=network"
+		],
+		"env": {
+			"RUST_BACKTRACE": "1",
+			"RUST_LOG": "ouch=debug"
+		}
+	},
+	"cleanup": [
+		"/include",
+		"/lib/pkgconfig",
+		"/man",
+		"/share/doc",
+		"/share/gtk-doc",
+		"/share/man",
+		"/share/pkgconfig",
+		"*.la",
+		"*.a"
+	],
+	"modules": [
+		{
+			"name": "blueprint-compiler",
+			"buildsystem": "meson",
+			"cleanup": [
+				"*"
+			],
+			"sources": [
+				{
+					"type": "git",
+					"url": "https://gitlab.gnome.org/jwestman/blueprint-compiler.git",
+					"tag": "v0.14.0"
+				}
+			]
+		},
+		{
+			"name": "ouch",
+			"builddir": true,
+			"buildsystem": "simple",
+			"build-commands": [
+				"blueprint-compiler batch-compile src/ui src/ui src/ui/window.blp src/ui/about.blp src/ui/about-shrimple.blp src/ui/help-overlay.blp src/ui/preferences.blp src/ui/oobe.blp",
+				"cargo build --release --features flatpak",
+
+				"mkdir -p ${FLATPAK_DEST}/share/locale/fr/LC_MESSAGES",
+				"mkdir -p ${FLATPAK_DEST}/share/locale/pt_BR/LC_MESSAGES",
+				"mkdir -p ${FLATPAK_DEST}/share/locale/nb_NO/LC_MESSAGES",
+				"mkdir -p ${FLATPAK_DEST}/share/locale/eo/LC_MESSAGES",
+				"msgfmt -o ${FLATPAK_DEST}/share/locale/fr/LC_MESSAGES/ouch.mo po/fr.po",
+				"msgfmt -o ${FLATPAK_DEST}/share/locale/pt_BR/LC_MESSAGES/ouch.mo po/pt_BR.po",
+				"msgfmt -o ${FLATPAK_DEST}/share/locale/nb_NO/LC_MESSAGES/ouch.mo po/nb_NO.po",
+				"msgfmt -o ${FLATPAK_DEST}/share/locale/eo/LC_MESSAGES/ouch.mo po/eo.po",
+
+				"install -Dm 644 res/${FLATPAK_ID}.gschema.xml --target-directory ${FLATPAK_DEST}/share/glib-2.0/schemas",
+				"glib-compile-schemas ${FLATPAK_DEST}/share/glib-2.0/schemas",
```

Comment:
so there is no build system?

Reply:
?

Reply:
yeah where is the build system. All of that belong to a build system upstream. Like virtually any other bit of software.

Reply:
we don't use a build system, but if it counts, we do use a justfile

Reply:
I don't know what a justfile is, but really all these build step should be part of a build system so that it consistent and easier for everyone.

Reply:
okay but `just` (the program that runs justfiles) is also a rust binary too, and you mentioned in the last review that internet access isn't allowed

Reply:
just provides binaries, look the the system76 cosmic baseapp packaging.

Reply:
https://github.com/flathub/com.system76.Cosmic.BaseApp/blob/bcd1a941a741715a6e831ed99138ae973c0e8796/com.system76.Cosmic.BaseApp.json#L8-L36

Reply:
> okay but `just` (the program that runs justfiles) is also a rust binary too, and you mentioned in the last review that internet access isn't allowed

also if i use a justfile, it wont use the `${FLATPAK_DEST}` variables, and i really don't want to maintain a separate justfile for these changes. i think that the way the flatpak builds is good enough for now until i can figure out a way to do this using just

Reply:
it should support setting a preifx which is like a 3 decades old common usage. And then you set the prefix to $FLATPAK_DEST before running it. That's a feature a standard build system provides. Anything hardcoded is just broken.